### PR TITLE
Search filters

### PR DESCRIPTION
**Pull request by tigermli**

### CODE REVIEW

File: search/views.py [hunk 1]
```diff
@@ -3,27 +3,135 @@
...
+        searchResults = Student.objects.none()
+
+        # query input check
+        query = self.request.query_params.get('query', None)
+        if (query is None or not query.isalnum()):
+            return list(searchResults) 
+
+        # query with profile filter
+        profileFilter = self.request.query_params.get('profileFilter', None)
+        if (profileFilter is not None):
```
Comment (jerrylinew) [inline search/views.py]: no need for parenthasis

File: search/views.py [hunk 2]
```diff
@@ -3,27 +3,135 @@
...
+            if (profileFilter == "Student"):
```
Comment (jerrylinew) [inline search/views.py]: ditto

File: search/views.py [hunk 3]
```diff
@@ -3,27 +3,135 @@
...
+
+                gradeFilter = self.request.query_params.get('gradeFilter', None)
+                if (gradeFilter is not None):
+                    try:
+                        gradeFilter = int(gradeFilter)
+                        if 1 <= gradeFilter and gradeFilter <= 12:
+                            searchResults = searchResults.filter(grade=gradeFilter)
+                    except ValueError:
+                        pass
+            elif (profileFilter == "Instructor"):
```
Comment (jerrylinew) [inline search/views.py]: ditto

File: search/views.py [hunk 4]
```diff
@@ -3,27 +3,135 @@
...
+                searchResults = Instructor.objects.search(query)
+            elif (profileFilter == "Admin"):
```
Comment (jerrylinew) [inline search/views.py]: ditto

File: search/views.py [hunk 5]
```diff
@@ -3,27 +3,135 @@
...
+        profileFilter = self.request.query_params.get('profileFilter', None)
+        if (profileFilter is not None):
+            if (profileFilter == "Student"):
+                searchResults = Student.objects.search(query);
+
+                gradeFilter = self.request.query_params.get('gradeFilter', None)
+                if (gradeFilter is not None):
+                    try:
+                        gradeFilter = int(gradeFilter)
+                        if 1 <= gradeFilter and gradeFilter <= 12:
```
Comment (jerrylinew) [inline search/views.py:39]: grade can be 13 (for already graduated students)

File: search/views.py [hunk 6]
```diff
@@ -3,27 +3,135 @@
...
+                searchResults = searchResults.order_by("start_date")
+            if (sortFilter == "dateDesc"):
+                searchResults = searchResults.order_by("-start_date")
+            if (sortFilter == "timeAsc"):
+                searchResults = searchResults.order_by("start_time")
+            if (sortFilter == "timeDesc"):
+                searchResults = searchResults.order_by("-start_time")
+        return list(searchResults)
+
+'''
```
Comment (jerrylinew) [inline search/views.py]: what's this?

File: search/views.py [hunk 7]
```diff
@@ -32,3 +140,4 @@ def get_queryset(self):
                 Course.objects.search(query))
             return list(queryset_chain)
         return list(Student.objects.none())
+'''
```
Comment (jerrylinew) [inline search/views.py]: i'm assuming this is not used anymore? feel free to delete this if deprecated

**Review by jerrylinew — APPROVED** [review]
looks mostly good, some small comments to fix

File: search/views.py (hunk 6 above)
Comment (tigermli) [reply search/views.py]: This is the filters for course search

**Pull request merged**